qubes-rpc/tar2qfile.c

```diff
@@ -762,7 +764,9 @@ ustar_rd (int fd, struct file_header * untrusted_hdr, char *buf, struct stat * s
 			dirs_headers_sent[n_dirs-1] = malloc(sizeof (char) * (strlen(path)+1));
 			if (dirs_headers_sent[n_dirs-1] == NULL)
 				return MEMORY_ALLOC_FAILED;
-			strncpy(dirs_headers_sent[n_dirs-1], path, strlen(path)+1);
+			
+			memcpy(dirs_headers_sent[n_dirs-1], path, strlen(path)+1);
```

Comment:
@marmarek: GCC8 complains about length depending on the source length. Forcing a '\0' null byte after seems good for you?